Libraries/MLXLLM/SuScaledRotaryEmbedding.swift

```diff
@@ -50,3 +55,12 @@ public class SuScaledRotaryEmbedding: Module {
         )
     }
 }
+
+// Extension to Embedding to support using it as a linear layer
+extension Embedding {
+    public func asLinear(_ x: MLXArray) -> MLXArray {
+        // Use the embedding weights as a linear layer
+        // This is equivalent to x @ weight.T
+        return matmul(x, weight.transposed())
+    }
+}
```

Comment:
The embedding layer already has that: https://github.com/ml-explore/mlx-swift/blob/main/Source/MLXNN/Embedding.swift#L39-L45

Can you remove that code?